main.js

```diff
@@ -16,143 +16,139 @@ const tokenHUDWildcard = {
             range: { min: 0, max: 100, step: 1 },
             type: Number,
             default: 50
-        })
+        });
     }
 }
 
 const getTokenDimensions = (token, imgName) => {
-    const height = imgName.match(/_height(.*)_/)
-    const width = imgName.match(/_width(.*)_/)
-    const scale = imgName.match(/_scale(.*)_/)
+    const height = imgName.match(/_height(.*)_/);
+    const width = imgName.match(/_width(.*)_/);
+    const scale = imgName.match(/_scale(.*)_/);
 
-    const prototypeData = token._actor.data.token
+    const prototypeData = token._actor.prototypeToken;
 
     return {
         height: height ? parseFloat(height[1]) : prototypeData.height,
         width: width ? parseFloat(width[1]) : prototypeData.width,
-        scale: scale ? parseFloat(scale[1]) : prototypeData.scale,
+        scale: scale ? parseFloat(scale[1]) : prototypeData.texture.scaleX,
     }
 }
 
 Hooks.on('init', () => {
-    tokenHUDWildcard.registerSettings()
-})
+    tokenHUDWildcard.registerSettings();
+});
 
 Hooks.on('ready', () => {
-    const renderTokenConfig = 'renderTokenConfig'
-    Hooks.on(renderTokenConfig, WildcardDefault.render)
-    WildcardDefault._hookPreTokenCreate()
-})
+    const renderTokenConfig = 'renderTokenConfig';
+    Hooks.on(renderTokenConfig, WildcardDefault.render);
+    WildcardDefault._hookPreTokenCreate();
+});
 
 const WildcardDefault = {
     getDefaultImg: async (token) => {
-        const flag = token.getFlag('token-hud-wildcard', 'default') || ''
-        return flag
+        const flag = token.getFlag('token-hud-wildcard', 'default') || '';
+        return flag;
     },
-    _hookPreTokenCreate () {
+    _hookPreTokenCreate() {
         Hooks.on('preCreateToken', (parent, data, options, userId) => {
-            const defaultValue = parent?.actor?.data?.token?.flags['token-hud-wildcard'] ? parent.actor.data.token.flags['token-hud-wildcard'].default : ''
-
-            if (defaultValue !== '' && parent?.actor?.data?.token?.randomImg) {
-                const dimensions = getTokenDimensions(parent, defaultValue)
-                let updateInfo = { img: defaultValue , ...dimensions }
-                mergeObject(data, updateInfo)
-                parent.data.update(updateInfo)
+            const defaultValue = parent?.actor?.prototypeToken?.flags['token-hud-wildcard'] ? parent.actor.prototypeToken.flags['token-hud-wildcard'].default : '';
+            if (defaultValue !== '' && parent?.actor?.prototypeToken?.randomImg) {
```

Comment:
This can be changed to...
```javascript
const defaultValue = parent?.actor?.prototypeToken?.getFlag('token-hud-wildcard', 'default');
if (defaultValue && parent?.actor?.prototypeToken?.randomImg) {
```
Just a style thing, but it makes the code much easier to read